typewriter/script.js

```diff
@@ -0,0 +1,64 @@
+// with the help of GPT #typewriter
+
+const techStack = ['Python', 'Flask', 'Javascript', 'HTML', 'CSS'];
+const techStackSpan = document.getElementById('typewriter');
+const pauseBetweenWords = 1500;
+const typingInterval = 150;
+let indexTechStack = 0;
+let indexWord = 0;
+
+function typeTechStack() {
+    if (indexWord < techStack[indexTechStack].length) {
+        // Add next character
+        techStackSpan.textContent += techStack[indexTechStack].charAt(indexWord);
+        indexWord++;
+        setTimeout(typeTechStack, typingInterval);
+    } else {
+        // Pause after typing a word
+        setTimeout(() => {
+            techStackSpan.textContent = ''; // Clear the span
+            indexWord = 0; // Reset character index
+            indexTechStack = (indexTechStack + 1) % techStack.length; // when last index modulo array length is zero = reset index
+            typeTechStack(); // Start typing the next word
+        }, pauseBetweenWords);
```

Comment:
This is inherently a bit hard to read because its doing something complicated. If I was being nitpicky about it I'd probably try to refactor it to pull the if and the else clauses into separate functions to make it more obvious what was happening but honestly it's probably fine.